core/templates/dev/head/components/ck-editor-helpers/ck-editor-4-rte.directive.ts

```diff
@@ -86,8 +80,10 @@ angular.module('oppia').directive('ckEditor4Rte', [
         }).join(',');
         var buttonNames = [];
         names.forEach(function(name) {
-          buttonNames.push('Oppia' + name);
-          buttonNames.push('-');
+          if (ContextService.canWriteToFs()) {
```

Comment:
This is too broad, right? You only want this check if componentDefn.requiresFs is true.

Reply:
This will then show the toolbar(button), which we don't want, right?.
But if we keep this statement, then in learner's view the `buttonNames` will be empty and the toolbar won't be visible as required.

Reply:
@seanlip , I think this part can cause a bit of confusion, or am I missing something?

Reply:
I think if we use `componentDefn.requiresFs` and `canWriteToFs`, we might have to use `contextIsExplorationPlayer` to special case the suggestion modal and hide the toolbar in this case.  @seanlip , Do you have any other approach in mind?

Reply:
That's a good point. I think we do need two booleans then:
- canAddOrEditComponents (this is populated based on: is the context an editor page)
- canWriteToFilesystem (this is based on canAddOrEditComponents + the type of component)

and configure the checks accordingly based on that.